iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -126,6 +136,26 @@ static LogicalResult setRootDefaultConfig(FuncOp entryPoint, Operation *op) {
       threadTileSizes[depth] = 0;
     }
   }
+
+  // Calculate the problem size to adjust the tile size.
+  int64_t problemSize = 1;
+  entryPoint.walk([&problemSize](IREE::Flow::DispatchTensorStoreOp storeOp) {
```

Comment:
Its a mere co-incidence that the problem size is same as the output size. Doing this blanket for default operations is ill-defined. If really needed, at least for now we should do this for cases where we are looking for specific "root ops".

Reply:
Maybe I'm not really understanding what problem size means then. What I really want is the output size so that I can pick the tile size based on the number of workgroup.

>If really needed, at least for now we should do this for cases where we are looking for specific "root ops".
I'm not sure if this is what you mean but this whole function is meant to be called for the root op only.

Could you give a bit more details on what you think is the problem? Otherwise we can discuss it later.

Reply:
This method here is for any op. In general, using the shape of the output as the problem size is not guaranteed. It happens to be the case for the operations that we typically see from MHLO/TOSA. Making that assumption in this method is a bit of a leap, and I am not sure how much this assumption will hold for `linalg_ext` ops. 

One example that comes to mind is `transpose` operations. If the indexing map for the output is an identity, then this assumption holds. Instead you could lower transpose with the input having identity indexing map. In that case, I am not sure this correlation holds. (In this case you are linearizing the dimensions, so it still holds, but hopefully this example gives you an idea of what I am saying are implicit assumptions made based on the current ops we see in IREE)

Thinking further, maybe you can add some additional checks that make sure that the correlation of problem size to output shape hold. I think checking that it is a `GenericOp` with all parallel iterator types and output indexing map being identity might be enough to guarantee problem size == output size.

Reply:
Makes sense, I see what you mean, thanks for explaining. I had originally limited it to `GenericOp` but somehow tried to generalize it, let me limit it for now.

Reply:
Changed the logic to only apply to generic ops with projected permutation output maps.